hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -2017,3 +2032,48 @@ func (r *NodePoolReconciler) doesCPODecompressAndDecodeConfig(ctx context.Contex
 	_, managesDecompressAndDecodeConfig := supportutil.ImageLabels(controlPlaneOperatorImageMetadata)[controlPlaneOperatorManagesDecompressAndDecodeConfig]
 	return managesDecompressAndDecodeConfig, nil
 }
+
+func (r *NodePoolReconciler) ensureMachineDeletion(nodePool *hyperv1.NodePool) error {
+	machineSetList := &unstructured.UnstructuredList{}
+	var gvk schema.GroupVersionKind
+	var err error
+
+	switch nodePool.Spec.Platform.Type {
+	// Define the desired template type and mutateTemplate function.
+	case hyperv1.AWSPlatform:
+		gvk, err = apiutil.GVKForObject(&capiaws.AWSMachineList{}, api.Scheme)
```

Comment:
we don't need anything of this. All we need to do is check there's no capi.Machines in the hcp namespace.

Reply:
Oh ok, I thought I would need to look at the specific CAPA instance to get the machines. Making the change now and testing.